Updating terminology page (#183)
risc0/website: docs/zkvm/key-terminology.md

# Key Terminology

### Circuit
[circuit]: #circuit
[Arithmetic Circuits]: ../reference-docs/about-arithmetic-circuits.md

> A mathematical construct that appears in one view as the "CPU" in the [zkVM] but on the other is part of the computation used to create [proofs]. <br/>

Reviewer: I'm struggling to understand what you mean by "appears in one view" and "the other" hear. Perhaps this rewording is clearer, if it still captures the key information you wanted to convey?
```suggestion
A mathematical construct that acts as the "CPU" of the [zkVM] in a manner that enables creating [proofs]. <br/>
```

See also: [Arithmetic Circuits]

### Commit
[commit]: #commit
[env::commit()]: https://docs.rs/risc0-zkvm/latest/risc0_zkvm/guest/env/fn.commit.html#
Append data to the [journal]. <br/>
See also: [env::commit()]

### Continuations
[Continuations study club]: https://www.youtube.com/watch?v=v4HIwaqmIxk&list=PLcPzhUaCxlCirUkJY0ltpjdtzWcz5U_6y&index=1
[Continuations blog]: https://risczero.com/blog/continuations
As of our 0.15 release, the [zkVM] uses continuations to enable proof generation for very large programs without blowing up memory requirements.
Large proof generation is made possible by splitting the execution trace for a single [session] into a number of [segments], and proving those segments independently. <br/>
See also: [Continuations study club], [Continuations blog]

### Execute

> Run a method inside the [zkVM] and produce a [receipt] that attests to correct execution. 

Reviewer: Executing is separate from proving; it produces an execution trace, not a receipt.
```suggestion
Run a method inside the [zkVM] and produce an [execution trace] which can be used by the [prover] to create a [proof] of correct execution. 
```

### Executor
The portion of the [zkVM] responsible for generating the [execution trace].

### Execution Trace
[execution trace]: #execution-trace
A complete record of a computation. 
The execution trace is organized as a rectangular array, where each row describes a complete snapshot of the state of the [zkVM] at a given moment in time.
The width of the execution trace relates to the number of registers/components in the machine, and the length of the execution trace relates to the number of clock cycles of the program's execution. 

### Guest
The system running inside the zkVM. <br/>
See also: [Rust crate for zkVM guest]

### Guest Program
[guest program]: #guest-program
The portion of a zkVM application that gets proven. 
The guest program receives inputs from an (untrusted) host. 
Depending on the application, the guest program can commit results to the journal and/or send them to the host. <br/>
See also: [Rust crate for zkVM guest]

### Host
[host]: #host
The system the zkVM runs on.

### Host program
The host-native, untrusted portion of a [zkVM] application. 

> The host program loads the guest program and provides inputs to the gust program as necessary. 

Reviewer: Typo
```suggestion
The host program loads the guest program and provides inputs to the guest program as necessary. 
```
